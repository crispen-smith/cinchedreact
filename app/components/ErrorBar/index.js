/**
 *
 * ErrorBar
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';
import colors from '../../resources/colors';

const EB = styled.div`
  width: auto;
  margin: 0.5rem 1rem;
  background-color: ${colors.bg_red};
  color: ${colors.fg_red};
  text-decoration: underline;
`;

const ErrorBar = props => <EB>{props.message}</EB>;

ErrorBar.propTypes = {
  message: PropTypes.string.isRequired,
};
export default ErrorBar;
