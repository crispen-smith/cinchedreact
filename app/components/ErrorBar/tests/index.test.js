import React from 'react';
import renderer from 'react-test-renderer';

import ErrorBar from '../index';

describe('<Errorbar />', () => {
  it('Renders an ErrorBar', () => {
    const tree = renderer.create(<ErrorBar message="TEST" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
