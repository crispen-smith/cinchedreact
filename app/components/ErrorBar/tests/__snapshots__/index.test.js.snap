// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Errorbar /> Renders an ErrorBar 1`] = `
<div
  className="ErrorBar-jIbckV kWZWPS"
>
  TEST
</div>
`;
